Sally7/Internal/NetworkOrderSerializer.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System;
+using System.Buffers.Binary;
+using System.Runtime.CompilerServices;
+
+namespace Sally7.Internal;
+
+internal static class NetworkOrderSerializer
+{
+    public static void WriteInt16(ref byte destination, short value)
```

Comment:
```suggestion
    public static void WriteInt16(ref byte destination, ushort value)
```
Found while inspecting the machine code (see other comment).
This was also sign-extended which isn't needed, as signed/unsigned does not make a difference in writing the bytes. So a little perf-boost here.

Either make the parameter unsigned, or cast to `ushort` in the two places used in this method.

Reply:
Final codegen should look like [this](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8gOgCEBXAMy5il3YBLAHbYoATwDcNek1YAlDsIyD8MFgGEI+AA6CANnwDKfAG6CwMXNOoyAzE1IMAYhAg0A3jQY+mDkRgMbGIAFLBcDMDiGDAMACZWGGgMAQw6Ytj4ADIwwgDmGAAWyalx2BjYOflFAJTevl7Uvs1MAOwMAOqCsK5Q+OUq+SwdUIIxAFIQwPIwAI4ciQASMNgJUGEwEQm4SWkZ2bkFxfHllYe1Ns0AvvU+t34pykFiAPoAksL6IjBxGxFRMXiiRKT3SUEyVSOIMCZQqkNq90aLV8xHaXR60H6GEGeWGowmUxm8yWKzW70+31+4SBO2SYIh52OsLO1UKNUuvhu1C5NACfFE+jkSEcnW6MF6WJxnnuOlGpnKsUgwh2j0CABUdABrQIAXgYdAQdDsLzodBNZtNNnuAHpbQwADy4Dj4foSAB8NrtABFTgwdlAOGAMCxPdaHdanS6xOIPU1fLaw/bYK7NbhY80Eww1YVBLg/RgA0GUnmIJ9xAwHACrAx9KW8skuCJsIKq8WGBBTHwAO74mLCENxnyZ+3W5NiVPp3yywTywFKlWpH0VBh6pz2+0MUgoBh2+GFe7NQj64DkU/kc2mh0b8jCu1vBLKQSNvgH3xHugn00Wr9XhgADkkHcwwABS9ABVBhhGdYA+AYbBhDiBgAFEAHk1RoGU5QVBh50CVJJmmOYFh2ZZVj4cgV31BA7DXDctyAhhgKgCAMAgSBBTeL1Xx8I8QirGoAFkrFwbA8hgNVxB0dQCKJYjAnXBgb0Au0hNwESxIYDBJJgDDByAxNI1dGNQyzQpYlAiDwj4XJLAHDM7RHQzo0nIcHLHKAJxM7NcwYeV9AWNszNgBguGgGsYDybAwHLWA8FLARTNiR0YEVcIdQAIi0YQwA4KBYGUWTEiQhAYBy1ioHS603QYfAIDiJ9BGrIpYlrLsXz0zNW1YlIHxULgYqsHR4qsZJmoYDgdCkqBImiWJc2EAByQIoowDhm30csOFwH44MCMbarUZQ7PjBzRxgFM00wmdsNwv1CmgQJgLiDgZgiPUKIUgDdOaYgHGYYVUhGMZxK1DA/hmwFtkGcpBFLaFwtZOo9KRZE2k6fEYDAj4MBo8GoabFQ4azUGGF4jgAhqQxEY5HweT036hVVdHgaXbBwdbfHREJ4QkeaFHkTtXo0j4UK+ngywGDayWMbg4QmZEHZSXbCJtvpQE/OI0azJCpsW1m5XNO11IuwMFtYg7bte1yY7kSBmIseUXHqU5mGidZ9lvtR1EKxphg6Z+v6byZu2YBkoiSTI9ZqQ5xICdh4R4fpA5WXh5k915hpuOZ+3sadzYaWh7nkjD4kSNJciPb0+yGBySLoq1nSq98EOHYwG88djrn45YABxGAMBQnhtrBlAamSD6Nz/dkGIshhwizlvsfb53O9d/s+4Hof+5CJAx4YEItvuqAMBqJP0995pF+UZf85d7ne/7weuGHkIp+SA/cCPk+08ZSus+98gdBfZcmaPcAA2kJIodU3i6H0CESB904gwJ0PoFCOhuYCAAIJ5DyLANSM4YAfC+MIEQeQagAF17gM3+sHDGJc5KkTJEQyk7N9Z33jonfYe5U6nHToiBeGNW55y2KvIuDB6ERzWOQSuqMGK1yiuIBuAjgatxviInYcdSwP03s/beo9x6/inspEC4E56bGUTna+SAO4aK7lojeT8X673fofB6p8uG/wvs3QRS9rEr1sWvbRjjt5v33q44+NQf7U09siABQD7j+xRIHAGTwr5tz8fnGOASxHhMCBrGAGcfD8xaAAOX7l2aAmoUJQDWCYUYzZBAAC8+B4hUb4mxhcOG+WbAsGRXt2i4CaTACAXAQifzccAmJDwaGAx8Y7UgrDIaiK6eTJ4+TCkMGKc0MpGAKkeWqbUvgggGnNKgK0yxOMFn+M6UTdZXifDe0Gc0kZIQKaTO5LpPkUABSMzIFmbSxU+yDPitKemySzH/H1g4reYMig+Wjmw5ZRNUgjN0SffhTcHntGpGBZU2BeAsEwXEKkt8kUJzCRTVFw8+m0xoHTL5PyaF/J2XsqpNTjBHJOe1Yp1Cg7EG3GktRENYjsKJrk7p/kCkYtkbikSBKW4CkEHkYQPwOmaITlnZobAxhZU7MfFpbxcBZDGBgQwSEELHLlgAfiCE2CQTFVBjAIQIGYertrmvqvBFVakQjrIYCACVvT3lUPBfy7OmNc5XMyYi7JKzUjrOlajWV+L1AKoacq1V1z1VoE1b4bVGBdV8BiGcw1xrsRmotfBBgNrtWiHtaMfATrOwupgG6mAHrLXetwL6npBT/WBoKe8+lyh+TNgYLkZ0DBVLqXEtpftVZQXNAkSqPUBo6DkBzXpTBYBNSUTXaQTdzRt2alZnuw0dhD2+DAttKAp7V2GlaHSoAA), so no more sign-extending moves -- they are slow and not needed.